Removing H preserving only wedged ones strips all H

This is probably not a big deal, because we'll probably also want to keep bond stereo defining H in this case, but this feels a bit embarrassing:

```python
smips = Chem.SmilesParserParams()
smips.removeHs = False
m = Chem.MolFromSmiles('FC([H])(O)Cl', smips)
m.GetBondBetweenAtoms(1, 2).SetBondDir(Chem.BondDir.BEGINWEDGE)
ps = Chem.RemoveHsParameters()

ps.removeWithWedgedBond = False;

ps.removeDefiningBondStereo = True;
ps.removeDegreeZero = True;
ps.removeDummyNeighbors = True;
ps.removeHigherDegrees = True;
ps.removeHydrides = True;
ps.removeInSGroups = True;
ps.removeIsotopes = True;
ps.removeMapped = True;
ps.removeNonimplicit = True;
ps.removeOnlyHNeighbors = True;
ps.removeWithQuery = True;

m = Chem.RemoveHs(m, ps)
assert m.GetNumAtoms() == 5

---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Input In [1], in <cell line: 22>()
     19 ps.removeWithQuery = True;
     21 m = Chem.RemoveHs(m, ps)
---> 22 assert m.GetNumAtoms() == 5

AssertionError: 
```

In any case, it's easy to fix. I'll post a patch in a minute.

--- a/Code/GraphMol/AddHs.cpp
+++ b/Code/GraphMol/AddHs.cpp
@@ -831,7 +831,7 @@ void removeHs(RWMol &mol, const RemoveHsParameters &ps, bool sanitize) {
     bool removeIt = true;
     if (atom->getDegree() &&
         (!ps.removeDummyNeighbors || !ps.removeDefiningBondStereo ||
-         !ps.removeOnlyHNeighbors)) {
+         !ps.removeOnlyHNeighbors || !ps.removeWithWedgedBond)) {
       bool onlyHNeighbors = true;
       ROMol::ADJ_ITER begin, end;
       boost::tie(begin, end) = mol.getAtomNeighbors(atom);

--- a/Code/GraphMol/molopstest.cpp
+++ b/Code/GraphMol/molopstest.cpp
@@ -8389,6 +8389,41 @@ M  END)CTAB"_ctab;
   BOOST_LOG(rdInfoLog) << "\tdone" << std::endl;
 }
 
+void testGithub5099() {
+  BOOST_LOG(rdInfoLog)
+      << "-----------------------\n Testing github issue 5099: "
+         "Removing H preserving only wedged ones strips all H"
+      << std::endl;
+
+  std::string smi{"FC([H])(O)Cl"};
+  std::unique_ptr<RWMol> m{SmilesToMol(smi, false, false)};
+  TEST_ASSERT(m);
+  TEST_ASSERT(m->getNumAtoms() == 5);
+  m->getBondBetweenAtoms(1, 2)->setBondDir(Bond::BondDir::BEGINWEDGE);
+
+  auto ps = MolOps::RemoveHsParameters();
+  ps.showWarnings = false;
+
+  // Remove all but Wedged/dashed H
+  ps.removeWithWedgedBond = false;
+  ps.removeDefiningBondStereo = true;
+  ps.removeDegreeZero = true;
+  ps.removeDummyNeighbors = true;
+  ps.removeHigherDegrees = true;
+  ps.removeHydrides = true;
+  ps.removeInSGroups = true;
+  ps.removeIsotopes = true;
+  ps.removeMapped = true;
+  ps.removeNonimplicit = true;
+  ps.removeOnlyHNeighbors = true;
+  ps.removeWithQuery = true;
+
+  removeHs(*m, ps);
+
+  // H shouldn't be removed
+  TEST_ASSERT(m->getNumAtoms() == 5);
+}
+
 int main() {
   RDLog::InitLogs();
   // boost::logging::enable_logs("rdApp.debug");
@@ -8504,6 +8539,7 @@ int main() {
   testGithub3854();
   testSetTerminalAtomCoords();
   testGet3DDistanceMatrix();
+  testGithub5099();
 
   return 0;
 }

